LWanjiru/le-documentaire
# #148963113 Add API Endpoints, Tests and Blueprint

*   **_What does this PR do?_**
  - Add controller methods
  - Configure API endpoints
  - Add tests for API testing
*   **_Description of Task to be completed?_**
  - Testing and creation of a RESTFUL API to be consumed by a React-Redux application
*   **_How should this be manually tested?_**
  1. Using postman to test the various Endpoints
    - use `$ nodemon` to run the server locally after cloning( instructions below).
   - use Postman to test all the endpoints in `server/routes/Routes.js`

  2.  Clone this repository locally using the command `$ git clone https://github.com/LWanjiru/le-documentaire.git`.
     - $ cd le-documentaire and use the command `$ git checkout feature/148963113/build-backend-with-tdd`
    - run `$ npm install`
    - use the commands `$ npm test` (with reporter only)  or `$npm run test:dev` (with both coverage .html and reporter)

*   **_Any background context you want to provide?_**
    - In order to fully interact with the API, one must have installed and set up a Database with [PostgreSQL](https://www.postgresql.org/)). And installed [sequelize](http://docs.sequelizejs.com/)


     - Ensure to migrate the models and seed the tables using the following commands:
    `$ sequelize db:migrate && sequelize db:seed:all`

     [Here are some other commands to help you along with sequelize.](https://github.com/sequelize/cli)

*   **_What are the relevant pivotal tracker stories?_**

   https://www.pivotaltracker.com/n/projects/2072191 
*   **_Screenshots (if appropriate)_**

<img width="809" alt="screenshot 2017-08-20 13 48 10" src="https://user-images.githubusercontent.com/27014312/29494225-560375fa-85ae-11e7-8824-90b233322661.png">
<img width="842" alt="screenshot 2017-08-20 13 47 53" src="https://user-images.githubusercontent.com/27014312/29494226-58f08c30-85ae-11e7-8cba-9f6c2bea0d91.png">
<img width="865" alt="screenshot 2017-08-20 13 48 33" src="https://user-images.githubusercontent.com/27014312/29494227-5c4d9d5a-85ae-11e7-8d8b-c66ef47d58c0.png">
